
## Z-Test Lab for Alternative Bus Route


In [1]:
from __future__ import print_function

import numpy as np
import pandas as pd
import pylab as pl
import scipy.stats

import os
import json

%pylab inline

Populating the interactive namespace from numpy and matplotlib


According to the instructions, I am using the Z test to understand if a (fictitious) implementation of an alternative bus route for (fictitious) bus line X8 improves circulation.

Idea: The alternative bus route is more efficient than the previous and saves trip time.

Null Hypothesis: The average trip duration of the alternative bus route is longer (greater) than the average trip duration of the old route with a significance level of 0.05.

Ho: Ta > To

Alternative Hypothesis: The average trip duration of the alternative route is less or equal than the average trip duration of the old route with a significance level of 0.05.

Ha: Ta <=To

Alpha = 0.05

In [2]:
#get raw data
trip_alt = pd.read_table("https://raw.githubusercontent.com/fedhere/PUI2018_fb55/master/Lab4_fb55/times.txt", header = None).values.squeeze()
trip_alt

array([ 31.62223931,  32.82137636,  30.2291014 ,  31.41376587,
        39.01055035,  34.82207891,  39.87188366,  39.57994562,
        31.02658678,  27.66246068,  18.8752884 ,  46.2313731 ,
        30.01522604,  47.6004104 ,  16.52726666,  42.4462331 ,
        33.62527449,  42.28462558,  46.20520667,  30.87517392,
        32.94800341,  26.17691512,  31.25699649,  35.33644554,
        35.64072501,  40.32297174,  38.04348588,  29.4579793 ,
        23.91636367,  34.56403521,  45.42782464,  38.95607754,
        39.11396748,  39.67398759,  31.89148105,  22.62379325,
        26.18910408,  43.54114213,  33.2607386 ,  31.67602384,
        37.12846259,  38.22048496,  34.38814185,  36.11407371,
        44.08064814,  29.59917161,  37.02411751,  41.60576556,
        44.37860528,  38.06471077,  22.90526456,  40.27182857,
        31.57979239,  25.88037795,  28.8913048 ,  44.33687925,
        36.78948684,  43.46098875,  40.65557   ,  26.16634004,
        25.13884784,  33.01581456,  32.87050073,  26.28

In [3]:
trip_alt.shape

(100,)

In [4]:
#the mean for alterntive route
altMean = trip_alt.mean()
altMean

34.466161688299998

In [5]:
#standard deviation for alterntive route
altStd = trip_alt.std()
altStd

7.1015040681937771

In [6]:
#the old bus rout for X8 took 36 +- 6 minutes (mean and standard deviation parameters of the population): 
popMean = 36
popStd = 6


In [7]:
# Running Z-test - how many standard deviations away from the mean

sqrt_N = np.sqrt(len(trip_alt))
Z = ((popMean - altMean) / (popStd / sqrt_N))
Z

2.5563971861666701

## Conclution: 

Since Z test resulted in a value of 2.56 > 2, which means that the alternative mean is further than 2 sigmas than the old mean, we can reject NULL hypothesis 

The alternative bus route is significantly more efficient and faster than the old bus route.